File: generated_code/phi4/baseline_code_java/code_row_103.java
import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

public class FileHandler {
    public static void main(String[] args) {
        String filePath = "example.txt";
        
        try (FileOutputStream fos = new FileOutputStream(filePath, false)) {
            // The file is opened for writing only.
            // If the file does not exist, it will be created.
            // If the file already exists, its content will be truncated to 0 bytes.
            System.out.println("File opened/created and truncated successfully.");
        } catch (IOException e) {
            if (e.getMessage().contains("exists")) {
                System.out.println("File already exists. Operation failed due to O_EXCL flag behavior.");
            } else {
                e.printStackTrace();
            }
        }
    }
}

File: generated_code/phi4/baseline_code_py/code_row_103.py
import os

def create_or_open_file(file_path):
    # Try to open a file with flags: O_CREAT | O_TRUNC | O_WRONLY | O_EXCL
    try:
        fd = os.open(file_path, os.O_CREAT | os.O_TRUNC | os.O_WRONLY | os.O_EXCL)
        print(f"File '{file_path}' created and opened for writing.")
        # Close the file descriptor after usage
        os.close(fd)
    except FileExistsError:
        print(f"File '{file_path}' already exists.")

# Example usage
create_or_open_file("example.txt")